retrodep/gosource.go

```diff
@@ -533,19 +533,23 @@ func (src GoSource) RepoPathForImportPath(importPath string) (*RepoPath, error)
 
 	// No replacement found, use the import pth as-is
 	r, err := vcs.RepoRootForImportPath(importPath, false)
-	if err != nil && strings.ContainsRune(importPath, '_') {
+	if err == nil {
+		return &RepoPath{RepoRoot: *r}, nil
+	} else {
```

Comment:
The convention in Go is to not `else` when the previous block ends in `return`, `continue`, etc. The `u == -1` doesn't have an `else` after its `return`.

Reply:
Yes. I think it would read better like this:
```
if err != nil {
    // Special handling only for paths containing "_".
    u := strings.Index(importPath, "_")
    if u == -1 {
        return nil, err
    }

    // go.pkg ... it and try again.
    importPath = path.Dir(importPath[:u])
    r, nerr := vcs.RepoRootForImportPath(importPath, false)
    if nerr != nil {
        return nil, err // Returning the initial error is intentional
    }
}
return &RepoPath{RepoRoot: *r}, nil
```

This way it's clear from the flow that we're in an unusual case, we handle both the calls, and we're only ever at two additional indentations (error handling within the error handling).

What do you think?

Reply:
Me? Yes, I think that's fine. OTOH returning early when the result is known, `err == nil`, avoids having to mentally stack the covering condition for what follows and is my inclination, OTOH _Effective Go_'s advice isn't _return early_ but _success flows down_, and folks are so used to `if`'s test of `err` being not-equal that an `==` can be skipped/misread at first glance.

If I had to nitpick, `nerr` implies a new error that's replacing the old; `err2` would just suggest another error. :-)

Reply:
if/else - Done
nerr -> err2 - Done